packages/wrangler/src/queues/client.ts

```diff
@@ -46,208 +48,248 @@ export type Consumer = ScriptReference & {
 	type: string;
 };
 
-export interface QueueResponse {
-	queue_id: string;
+export interface TypedConsumerResponse extends Consumer {
 	queue_name: string;
 	created_on: string;
-	modified_on: string;
-	producers: ScriptReference[];
-	producers_total_count: number;
-	consumers: Consumer[];
-	consumers_total_count: number;
-	settings?: QueueSettings;
+}
+
+export interface PostTypedConsumerBody {
+	type: string;
+	script_name?: string;
+	environment_name?: string;
+	settings: ConsumerSettings;
+	dead_letter_queue?: string;
+}
+
+export interface ConsumerSettings {
+	batch_size?: number;
+	max_retries?: number;
+	max_wait_time_ms?: number;
+	max_concurrency?: number | null;
+	visibility_timeout_ms?: number;
+	retry_delay?: number;
+}
+
+export class QueuesUrlBuilder {
+	private url: string;
+	private hasQueueId: boolean;
+	constructor(accountId: string) {
+		this.url = `/accounts/${accountId}/queues`;
+		this.hasQueueId = false;
+	}
+
+	withQueueId(queueId: string): QueuesUrlBuilder {
+		this.url += `/${queueId}`;
+		this.hasQueueId = true;
+		return this;
+	}
+
+	withConsumer(consumerId?: string): QueuesUrlBuilder {
+		if (!this.hasQueueId) {
+			throw new UserError(
+				"Can't access consumers without a queueId specified."
+			);
+		}
+		this.url += "/consumers";
+		if (consumerId) {
+			this.url += `/${consumerId}`;
+		}
+		return this;
+	}
+
+	build(): string {
```

Comment:
if the invocations of the `withConsumer` and `withQueueId` methods are only ever chained with `build()` - can we eliminate the build method and just return `this.url` from the methods that update it?

Reply:
`withQueueId` can be optionally chained with `withConsumer`, that's why I added a "return" method that actually returns the fully formed string.